### PR TITLE
EWL 5184: form tweaks

### PR DESCRIPTION
**Pull request by nstriedinger**
## Ticket(s)
https://issues.ama-assn.org/browse/EWL-5184
https://issues.ama-assn.org/browse/EWL-5228
https://issues.ama-assn.org/browse/EWL-5187
https://issues.ama-assn.org/browse/EWL-5189
https://issues.ama-assn.org/browse/EWL-5192
https://issues.ama-assn.org/browse/EWL-5194
https://issues.ama-assn.org/browse/EWL-5196
https://issues.ama-assn.org/browse/EWL-5206
https://issues.ama-assn.org/browse/EWL-5208

## Description
Go through all forms at `http://ama-one.local/admin/structure/webform` and verify the form inputs look like the following: http://localhost:3000/?p=viewall-atoms-forms

**We don't have a "page" layout for forms.**

**NOTE: I'm getting JS Errors on the Drupal side that need to be fixed.**

**BUG REPORT: https://issues.ama-assn.org/browse/EWL-5947**

## To Test:
- Run styleguide branch: `bugfix/EWL-5184-form-tweaks` https://github.com/AmericanMedicalAssociation/ama-style-guide-2/pull/403/
- Visit `http://localhost:3000/?p=viewall-atoms-forms`

### Drupal (develop branch)
- Make sure your provisioning/vars/local.yml is setup to use local styleguide.
- Run `composer install` (get updated webforms)
- Run `drush @one.local config-import -y && drush @one.local cr`

- Visit **`http://ama-one.local/form/stateregional-chair-registration`** and verify all the fields from here --> `https://www.ama-assn.org/eform/submit/state-or-regional-chair-registra`

- Visit **`http://ama-one.local/form/gov-council-action-item-request`** and verify all the fields from here --> https://www.ama-assn.org/eform/submit/governing-council-action-item

- Visit **`http://ama-one.local/form/fax-opt-out-form`** and verify all the fields from here --> https://www.ama-assn.org/eform/submit/fax-opt-out-form

- Visit **`http://ama-one.local/form/end-user-license-agreement`** and verify all the fields from here --> https://www.ama-assn.org/eform/submit/end-user-license-agreement-cpt

- Visit **`http://ama-one.local/form/cpt-network-registration`** and verify all the fields from here --> https://www.ama-assn.org/eform/submit/cpt-network-registration

- Visit **`http://ama-one.local/form/contact-us`** and verify it looks like `https://app.zeplin.io/project/59dd1f9c36a8b21d569bb9e8/screen/5acce9cdf2c8b8e263e1e04e`

---

[Guidelines for Contribution](CONTRIBUTING.md)
[SG2 Standards](ama-style-guide-2/docs/standards.md)

### CODE REVIEW

File: styleguide/source/assets/js/form-items.js [hunk 1]
```diff
@@ -168,23 +169,19 @@
...
+          // I can't find where we're using this and it errors on Drupal end.
+          // function selChange(){
+          //   var selection = $('#myCheckList').checkList('getSelection');
+          //   $('#selectedItems').text(JSON.stringify(selection));
+          // }
+          //
+          // $('#filterList').checkList({
+          //   listItems: dataModel,
+          //   onChange: selChange
+          // });
```
Comment (damontgomery) [inline styleguide/source/assets/js/form-items.js]: Let's ask the team if they know about this. I think I heard someone mention checkboxes the other day.

File: styleguide/source/assets/js/init.js [hunk 2]
```diff
@@ -25,12 +25,4 @@
...
 	};
 
-	Drupal.behaviors.jumpMenu = {
-		attach: function (context, settings) {
-			$('.js-dropdown-select').on('change', function () {
-				window.location = $(this).find(':selected').data('url');
-			});
-		}
-	};
-
```
Comment (damontgomery) [inline styleguide/source/assets/js/init.js:35]: Let's ask the team if this is used and remove / or add a conditional statement to see if there is a URL first.

File: styleguide/source/assets/js/webforms.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+ *
+ * JavaScript should be made compatible with libraries other than jQuery by
+ * wrapping it with an "anonymous closure". See:
+ * - https://drupal.org/node/1446420
+ * - http://www.adequatelygood.com/2010/3/JavaScript-Module-Pattern-In-Depth
+ */
+(function ($, Drupal) {
+
+  var verifyFields = function(form) {
+    console.log("hello");
```
Comment (damontgomery) [inline styleguide/source/assets/js/webforms.js]: Remove debug statement. :)

File: styleguide/source/assets/js/webforms.js [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    var $sections = form.find('section');
+    var $inputs = $('.webform-submission-form section *').filter(':input');
+    var $iconElement = $('.ama__form-steps__icon');
+    var errorSections = [];
+    var errorMessage = $('.form-item--error-message');
```
Comment (damontgomery) [inline styleguide/source/assets/js/webforms.js]: Are you using the $ = jQuery object? If so, I like that pattern and we should do this for `errorMessage` and others. Give it another check.

File: styleguide/source/assets/js/form-items.js (hunk 1 above)
Comment (nstriedinger) [reply styleguide/source/assets/js/form-items.js]: Yeah, I get errors on the Drupal side!

File: styleguide/source/assets/js/webforms.js (hunk 4 above)
Comment (nstriedinger) [reply styleguide/source/assets/js/webforms.js]: Yes, I am. I actually didn't need that variable. Good Catch!

File: styleguide/source/assets/js/webforms.js (hunk 3 above)
Comment (nstriedinger) [reply styleguide/source/assets/js/webforms.js]: I did.

File: styleguide/source/assets/js/init.js (hunk 2 above)
Comment (nstriedinger) [reply styleguide/source/assets/js/init.js:35]: ok

**Review by labbydev — APPROVED** [review]

**Pull request merged**